[Mix Transformer] Add Presets for MiTB0...MiTB5

DavidLandup0 opened the pull request:
This PR adds the presets for the MiT family.

- MiTB0 (3,321,962 params)
- MiTB1 (13,156,554 params)
- MiTB2 (24,201,418 params)
- MiTB3 (44,077,258 params)
- MiTB4 (60,847,818 params)
- MiTB5 (81,448,138 params)

Includes weight porting script from original SegFormer weights, the weights and configs are uploaded to Kaggle.
The weights are derived from the encoders fine-tuned on ADE20k and Cityscapes.

### Conversion Script

Run the conversion script as:

```
python tools/checkpoint_conversion/convert_mix_transformer.py --preset "B0_ade_512"
```

### Sanity Check

Mini notebook showing a quick training run on ImageNette with random weights vs using pre-trained weights as a quick sanity check for weight porting: https://colab.research.google.com/drive/10bod9H0lITVTDtyWnmezt5bwcDDXpfGw?usp=sharing

Accuracy plot comparing the two:

![image](https://github.com/user-attachments/assets/8bb40651-a56d-4897-b3fb-bc6d3d3d385c)

divyashreepathihalli commented:
Hi @DavidLandup0 here are some of the things I need for this PR

- [ ] A weights conversion script- to be added here - https://github.com/keras-team/keras-hub/tree/master/tools/checkpoint_conversion
- [ ] Weights to be uploaded here - https://www.kaggle.com/organizations/kerashub you already have an invite to join this org
- [ ] A demo Colab to show that the ported weights are working
- [ ] Weights have to be saved after including teh preprocessor flow has been added to the model - example follow resnet - https://github.com/keras-team/keras-hub/tree/master/keras_hub/src/models/resnet

DavidLandup0 commented:
@divyashreepathihalli Thank you for the checklist! 
We may actually have to update a layer or two of the model itself to port weights from HuggingFace. I can bundle that as part of this PR, actually. Would that be okay?

That way, we'd have the checkpoint conversion, model updates and presets in one place.

divyashreepathihalli commented:
Yes!! that sounds good!!

DavidLandup0 commented:
@divyashreepathihalli ready for review. Added:
- Weight porting script (supports ADE20k and Cityscapes as source weights)
- Presets for ADE20K and Cityscape MiTs B0...B5
- Preprocessor workflow, converter, etc.
- Uploaded weights to Kaggle
- Updated demo colab

divyashreepathihalli commented on keras_hub/src/models/mix_transformer/mix_transformer_presets.py:
```diff
@@ -0,0 +1,153 @@
+# Copyright 2024 The KerasHub Authors
```
remove copyright banner

divyashreepathihalli commented on keras_hub/src/utils/preset_utils.py:
```diff
@@ -155,7 +155,9 @@ def get_file(preset, path):
         try:
-            return kagglehub.model_download(kaggle_handle, path)
+            return kagglehub.model_download(
+                kaggle_handle, path, force_download=True
```
let us not set the `force_download` to True

DavidLandup0 replied:
Whoops, wasn't meant to be committed. Removed

divyashreepathihalli reviewed (commented):
LGTM!
just 2 nit comments
